kotsadm/pkg/handlers/config.go

```diff
@@ -447,49 +447,52 @@ func decrypt(input string, cipher *crypto.AESCipher) (string, error) {
 }
 
 func getLaterVersions(versionedApp *apptypes.App, startSequence int64) (int64, []versiontypes.AppVersion, error) {
-	versions, err := version.GetVersions(versionedApp.ID)
+	thisAppVersion, err := store.GetStore().GetAppVersion(versionedApp.ID, startSequence)
 	if err != nil {
-		return -1, nil, errors.Wrap(err, "failed to get app versions")
+		return -1, nil, errors.Wrap(err, "failed to get this appversion")
 	}
 
-	thisUpdateCursor := -1
-	latestSequenceWithUpdateCursor := startSequence
-	laterVersions := map[int]versiontypes.AppVersion{}
-	for _, version := range versions {
-		if version.Sequence == startSequence {
-			thisUpdateCursor = version.UpdateCursor
-		}
-	}
-	if thisUpdateCursor == -1 {
-		err := fmt.Errorf("unable to find update cursor for sequence %d in %+v", startSequence, versions)
-		return -1, nil, err
+	laterAppVersions, err := store.GetStore().GetAppVersionsAfter(versionedApp.ID, startSequence)
+	if err != nil {
+		return -1, nil, errors.Wrap(err, "failed to get later app versions")
 	}
-	for _, version := range versions {
-		if version.UpdateCursor == thisUpdateCursor && version.Sequence > latestSequenceWithUpdateCursor {
-			latestSequenceWithUpdateCursor = version.Sequence
+
+	// latestSequenceWithThisUpdateCursor is the newest local version
+	// of the same upstream version
+	latestSequenceWithThisUpdateCursor := thisAppVersion.Sequence
```

Comment:
`thisAppVersion` can be nil

Reply:
changed so that `thisAppVersion` cannot be nil